dashboard/app/controllers/api/v1/ml_models_controller.rb

```diff
@@ -2,14 +2,16 @@
 require 'cdo/aws/s3'
 
 class Api::V1::MlModelsController < Api::V1::JsonApiController
-  skip_before_action :verify_authenticity_token
   before_action :authenticate_user!
 
   S3_BUCKET = 'cdo-v3-trained-ml-models'
 
   # POST api/v1/ml_models/save
   # Save a trained ML model to S3 and a reference to it in the database.
   def save
+    # Setting CSRF token header allows us to access the token manually in subsequent POST requests.
+    headers['csrf-token'] = form_authenticity_token
```

Comment:
Since we're already in the POST request, does this actually do anything meaningful for validating the authenticity?

Reply:
+1 -- i would add this token to the GET request that loads whichever page needs access to this token for POST requests.

not sure if this helps, but i recently added this token to `appOptions` when script_levels are loaded in [this PR](https://github.com/code-dot-org/code-dot-org/pull/38182/files), so the client has access to the token via `appOptions.authenticityToken` from any script_level